src/builder/plugins/esbuild.ts

```diff
@@ -78,10 +85,10 @@ export function esbuild(options: Options): Plugin {
       }
 
       const result = await transform(code, {
-        ...options,
+        ...esbuildOptions,
         loader,
         sourcefile: id,
-        sourcemap: options.sourceMap ?? options.sourceMap,
+        sourcemap: sourceMap,
```

Comment:
```suggestion
        sourcemap: esbuildOptions.sourcemap ?? sourceMap,
```

Reply:
@ntnyq  maybe `sourceMap ?? esbuildOptions.sourcemap`  to keep `sourceMap `  prior , and be consistent with the current